src/Components/ServiceScene/Breakdowns/SingleViewTableScene.tsx

```diff
@@ -0,0 +1,235 @@
+              style={{
+                width: 'calc(100vw - 640px)',
+                minWidth: '200px',
+                textOverflow: 'ellipsis',
+                overflow: 'hidden',
+              }}
+            >
+              {props.cell.row.original.pattern}
+            </div>
+          );
+        },
+      },
+      {
+        id: 'include',
+        header: undefined,
+        disableGrow: true,
+        cell: (props: CellProps<WithCustomCellData>) => {
+          const existingPattern = appliedPatterns?.find(
+            (appliedPattern) => appliedPattern.pattern === props.cell.row.original.pattern
+          );
+          if (existingPattern?.type !== 'include') {
+            return (
+              <Button
+                variant={'secondary'}
+                fill={'outline'}
+                size={'sm'}
+                onClick={() => {
+                  props.cell.row.original.includeLink();
+                }}
+              >
+                Select
+              </Button>
+            );
+          }
+          return <></>;
+        },
+      },
+      {
+        id: 'exclude',
+        header: undefined,
+        disableGrow: true,
+        cell: (props: CellProps<WithCustomCellData>) => {
+          const existingPattern = appliedPatterns?.find(
+            (appliedPattern) => appliedPattern.pattern === props.cell.row.original.pattern
+          );
+          if (existingPattern?.type !== 'exclude') {
+            return (
+              <Button
+                variant={'secondary'}
+                fill={'outline'}
+                size={'sm'}
+                onClick={() => props.cell.row.original.excludeLink()}
+              >
+                Exclude
+              </Button>
+            );
+          }
+          return <></>;
+        },
+      },
+    ];
+    return columns;
+  }
+
+  /**
+   * Filter visible patterns in table, and return cell data for InteractiveTable
+   * @param patternFrames
+   * @param legendSyncPatterns
+   * @private
+   */
+  private buildTableData(patternFrames: PatternFrame[], legendSyncPatterns: Set<string>): WithCustomCellData[] {
+    const logExploration = sceneGraph.getAncestor(this, IndexScene);
+    return patternFrames
+      .filter((patternFrame) => {
+        return legendSyncPatterns.size ? legendSyncPatterns.has(patternFrame.pattern) : true;
+      })
+      .map((pattern: PatternFrame) => {
+        return {
+          dataFrame: pattern.dataFrame,
+          pattern: pattern.pattern,
+          sum: pattern.sum,
+          includeLink: () =>
+            onPatternClick({
+              pattern: pattern.pattern,
+              type: 'include',
+              indexScene: logExploration,
+            }),
+          excludeLink: () =>
+            onPatternClick({
+              pattern: pattern.pattern,
+              type: 'exclude',
+              indexScene: logExploration,
+            }),
+        };
+      });
+  }
+}
+
+function getVizStyles() {
```

Comment:
Nitpicky, but this can just be a `const styles = {}` without the function.

Reply:
Fixed!